CMakeLists.txt

```diff
@@ -0,0 +1,150 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2013-2022 The Foundry Visionmongers Ltd
+
+#-----------------------------------------------------------------------
+# CMake preamble.
+
+# FindPython's Development.Module component was added in 3.18, and
+# pybind11 recommends 3.18.2+.
+cmake_minimum_required(VERSION 3.18.2)
+
+# Additional include directories for CMake utils.
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+if (APPLE AND NOT DEFINED CMAKE_OSX_DEPLOYMENT_TARGET)
+    # The value of this variable should be set prior to the first
+    # project() command invocation because it may influence
+    # configuration of the toolchain and flags.
+    set(CMAKE_OSX_DEPLOYMENT_TARGET "10.13" CACHE STRING "Minimum OS X deployment version")
+endif ()
+
+# Forbid in-source builds.
+set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)  # NB: currently undocumented.
+# Forbid CMake from editing sources
+set(CMAKE_DISABLE_SOURCE_CHANGES ON)  # NB: currently undocumented.
+
+# By default, do not output "Up-to-date" messages during install. Can
+# be spammy when this project is included in another via CMake's
+# ExternalProject.
+if (NOT DEFINED CMAKE_INSTALL_MESSAGE)
+    set(CMAKE_INSTALL_MESSAGE LAZY)
+endif ()
+
+
+#-----------------------------------------------------------------------
+# Project metadata
+
+string(CONCAT project_description
+    "An open-source interoperability standard for tools and content"
+    " management systems used in media production.")
+
+project(
+    OpenAssetIO
+    VERSION 0.0.0
+    DESCRIPTION ${project_description}
+    HOMEPAGE_URL https://thefoundryvisionmongers.github.io/OpenAssetIO
+    LANGUAGES C CXX)
```

Comment:
LANGUAGES is optional, I'm not sure there's much benefit in documenting that its a C and C++ code-base here.

Reply:
Doing so does make CMake check that the compilers "work" for both languages as part of the preamble.

Reply:
Is there any C code though? I don't see any from a speed glance downward.

Reply:
> Doing so does make CMake check that the compilers "work" for both languages as part of the preamble.

From https://cmake.org/cmake/help/latest/command/project.html
> Optional. By default C and CXX are enabled if no language options are given.

Reply:
Interesting, the open source projects I'd used as a reference all seem to set this. I really don't know if there is any benefit.  I thought perhaps if/when we have C bindings then it helps document that.  Happy to remove it though.

Reply:
My last comment was written concurrently with both your replies, so I didn't see that.  Good to know.  I guess we can remove it then.

Reply:
Is `explicit` generally better than `implicit` from a maintenance perspective?

Reply:
Personally, I've never specified which languages I need, because the defaults have always covered me.

Do you really want to specify C and then have no C sources / maintain those definitions based on the source files available? For me, its less maintenance to not specify.

Reply:
Oh sorry, I thought you were saying the defaults assumed there _were_ C sources by default which would be adding undue checks, so we should put `LANGUAGES CXX` if that's all we have.

Reply:
After further discussion, including in the working group, we'll pretty much definitely have some C code in here.

I've left `LANGUAGES` in as it was.